spec/models/course/condition/achievement_spec.rb

```diff
@@ -20,11 +22,29 @@
 
       context "when an achievement is already included in its conditional's conditions" do
         subject do
-          create(:course_condition_achievement, :duplicate_child, course: course).tap do |cca|
+          create(:course_condition_achievement, course: course).tap do |cca|
             allow(cca).to receive(:achievement_id_changed?).and_return(true)
           end
         end
-        it { is_expected.to_not be_valid }
+
+        it 'is not valid' do
+          expect(subject).to_not be_valid
+          expect(subject.errors[:achievement]).to_not be_blank
+        end
+      end
+
+      context 'when an achievement is required by another conditional with the same id' do
```

Comment:
Actually I just recall that this test is to catch the mistake in the SQL query fixed in this PR [here](https://github.com/Coursemology/coursemology2/pull/693/files#diff-bf3adb0382fd1dc7ad1781f2636a1007L40). Please see the comment over there.

Adding the index will le us remove the validation of duplicated specific conditions though.